test/tstDetailsTreeNodeLabelling.cpp

```diff
@@ -0,0 +1,204 @@
+#include <ArborX_LinearBVH.hpp>
+
+#include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
+
+#include <random>
+
+template <typename DeviceType, typename T>
+auto toView(std::vector<T> const &v, std::string const &lbl = "")
+{
+  Kokkos::View<T *, DeviceType> view(lbl, v.size());
+  Kokkos::deep_copy(view, Kokkos::View<T const *, Kokkos::HostSpace,
+                                       Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+                              v.data(), v.size()));
+  return view;
+}
+
+template <class MemorySpace>
+struct MockBVH
+{
+  Kokkos::View<Kokkos::pair<int, int> *, MemorySpace> children_;
+  KOKKOS_FUNCTION auto size() const { return children_.extent(0) + 1; }
+};
+
+#define HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(MEMORY_SPACE)           \
+  template <>                                                                  \
+  auto ArborX::Details::HappyTreeFriends::getLeftChild<MockBVH<MEMORY_SPACE>>( \
+      MockBVH<MEMORY_SPACE> const &x, int i)                                   \
+  {                                                                            \
+    return x.children_(i).first;                                               \
+  }                                                                            \
+  template <>                                                                  \
+  auto                                                                         \
+  ArborX::Details::HappyTreeFriends::getRightChild<MockBVH<MEMORY_SPACE>>(     \
+      MockBVH<MEMORY_SPACE> const &x, int i)                                   \
+  {                                                                            \
+    return x.children_(i).second;                                              \
+  }
+
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(Kokkos::HostSpace)
+#ifdef KOKKOS_ENABLE_CUDA
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(Kokkos::Cuda::memory_space)
+#endif
+#ifdef KOKKOS_ENABLE_HIP
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(
+    Kokkos::Experimental::HIP::memory_space)
+#endif
+#ifdef KOKKOS_ENABLE_SYCL
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(
+    Kokkos::Experimental::SYCL::memory_space)
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(
+    Kokkos::Experimental::OpenMPTarget::memory_space)
+#endif
+
+namespace Test
+{
+template <class ExecutionSpace>
+auto findParents(ExecutionSpace const &exec_space,
+                 std::vector<Kokkos::pair<int, int>> const &children_host)
+{
+  auto children = toView<ExecutionSpace>(children_host, "Test::children");
+
+  auto const n = children.extent(0) + 1;
+  Kokkos::View<int *, ExecutionSpace> parents(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Test::parents"),
+      2 * n - 1);
+  Kokkos::deep_copy(parents, -1);
+
+  using ArborX::Experimental::findParents;
+  findParents(exec_space, MockBVH<Kokkos::HostSpace>{children}, parents);
```

Comment:
Why is it ` HostSpace`? Shouldn't it be `typename ExecutionSpace::memory_space`?

Reply:
You are right this is a bug